src/groups/mqb/mqbnet/mqbnet_multirequestmanager.t.cpp

```diff
@@ -235,48 +228,34 @@ class TestContext {
 };
 
 TestContext::TestContext(int nodesCount, bslma::Allocator* allocator)
-: d_blobBufferFactory(1024, allocator)
+: d_allocator_p(bslma::Default::allocator(allocator))
+, d_tempDir(d_allocator_p)
+, d_cluster_mp(new(*d_allocator_p) mqbmock::Cluster(
+                   d_allocator_p,
+                   true,   // isClusterMember
+                   false,  // isLeader
+                   false,  // isCSLMode
+                   false,  // isFSMWorkflow
+                   false,  // doesFSMwriteQLIST
+                   generateNodeDefs(nodesCount, d_allocator_p),
+                   "testCluster",
+                   d_tempDir.path()),
+               d_allocator_p)
+, d_nodes(clusterNodes(d_cluster_mp.get(), d_allocator_p), d_allocator_p)
 , d_blobSpPool_sp(
-      bmqp::BlobPoolUtil::createBlobPool(&d_blobBufferFactory,
-                                         bmqtst::TestHelperUtil::allocator()))
-, d_requestManager(0)
-, d_multiRequestManager(0)
-, d_requestContextSp(0)
-, d_cluster_mp(0)
-, d_nodes(allocator)
-, d_tempDir(allocator)
-, d_allocator_p(allocator)
+      bmqp::BlobPoolUtil::createBlobPool(d_cluster_mp->bufferFactory(),
+                                         d_allocator_p))
+, d_requestManager(
+      bsl::make_shared<ReqManagerType>(bmqp::EventType::e_CONTROL,
+                                       d_blobSpPool_sp.get(),
+                                       &d_cluster_mp->_scheduler(),
+                                       false,  // lateResponseMode
+                                       d_allocator_p))
+, d_multiRequestManager(
+      bsl::make_shared<MultiReqManagerType>(d_requestManager.get(),
+                                            d_allocator_p))
+, d_requestContextSp(d_multiRequestManager->createRequestContext())
```

Comment:
Here I reordered fields so they are constructed in the order of their dependencies and destructed in the reversed order.